hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/connector/map/SpecificPartitionsImapReaderPms.java

```diff
@@ -48,20 +51,45 @@
  */
 public abstract class SpecificPartitionsImapReaderPms<F extends CompletableFuture, B, R>
         extends LocalProcessorMetaSupplier<F, B, R> {
+    private final List<List<Expression<?>>> requiredPartitionsExprs;
     private transient int[] partitionsToScan;
     private transient Map<Address, int[]> partitionAssignment;
 
     private SpecificPartitionsImapReaderPms(
-            final BiFunctionEx<HazelcastInstance, InternalSerializationService, Reader<F, B, R>> readerSupplier) {
+            final BiFunctionEx<HazelcastInstance, InternalSerializationService, Reader<F, B, R>> readerSupplier,
+            @Nullable final List<List<Expression<?>>> requiredPartitionsExprs) {
         super(readerSupplier);
+        this.requiredPartitionsExprs = requiredPartitionsExprs;
     }
 
     @Override
     public void init(@Nonnull Context context) throws Exception {
         super.init(context);
-        Set<Integer> partitions = context.jobConfig().getArgument(KEY_REQUIRED_PARTITIONS);
-        if (partitions != null) {
-            partitionsToScan = partitions.stream().sorted().mapToInt(i -> i).toArray();
+        if (requiredPartitionsExprs != null) {
+            List<Integer> partitionsToScanList = new ArrayList<>();
+            partitionsToScan = new int[requiredPartitionsExprs.size()];
+            int j = 0;
+
+            HazelcastInstance hazelcastInstance = context.hazelcastInstance();
```

Comment:
this logic repeats (similar may be possibly for index scan), in the future potential candidate for refactoring